### DISCUSSION TITLE
Make transparent pixels opaque without losing alpha channel?

### DISCUSSION THREAD

**Discussion by violetgynoid** [ImageMagick/ImageMagick#6589] [Help]
### ImageMagick version

7.1.1

### Operating system, version and so on

Windows 10 22H2

### Description

I'm wanting to batch quantize images (to make them look a little more convincing for a DOOM mod while still bypassing the palette), and I found ImageMagick to do an excellent job! However, I'm having a hard time as a newbie figuring out a set of commands that would let me make transparent pixels over a certain threshold opaque, but not anything below that threshold. 

Would my best bet be something like applying a background color to an image and then using `-transparent`?

**Comment by fmw42** [comment]
Try this command where threshold is 250 out of 255.

 magick input_image ^
 ( +clone -alpha extract -fx "u>250/255?1:u" ) ^
 -alpha off -compose copy_opacity -composite ^
output_image

**Comment by snibgo** [comment]
A faster alternative is apply the `-fx` to just the alpha channel. This removes the need to extract the alpha, and copy it back:
```
magick input_image ^
-channel A -fx "u>250/255?1:u" +channel ^
output_image
```
This needs to calculate 250/255 for every pixel, which is a waste of time. We can optimize this:
```
magick input_image ^
-channel A -fx "u>%[fx:250/255]?1:u" +channel ^
output_image
```
Now it calculates 250/255 only once, and uses the result at each pixel.

(If the command is in a BAT script, double the `%`.)

**Reply by fmw42** [reply]
Of course one could also just define a variable T=250/255 and use the variable in place of 250/255 in the first equation.
